pubsub/index.js

```diff
@@ -68,35 +74,46 @@ const extractInfoFromMessage = (message) => {
   };
 };
 
-// Get attachment of a message..
+// Get attachment of a message.
 const extractAttachmentFromMessage = async (email, messageId, attachmentId) => {
-  await gmail.users.messages.attachments.get({
+  return gmail.users.messages.attachments.get({
     id: attachmentId,
     messageId: messageId,
     userId: email
   });
 };
 
-// Upload attachment of a message to Cloud Storage.
-const uploadAttachment = async (data, filename) => {
-  let file = storageClient.bucket(BUCKET).file(filename);
-  let writeableStream = file.createWriteStream({resumable: false});
-  writeableStream.write(data);
-  writeableStream.end();
+// Tag the attachment using Cloud Vision API
+const analyzeAttachment = async (data, filename) => {
+  var topLabels = ['', '', ''];
+  if (filename.endsWith('.png') || filename.endsWith('.jpg')) {
+    const [analysis] = await visionClient.labelDetection({
+      image: {
+        content: Buffer.from(data, 'base64')
+      }
+    });
+    const labels = analysis.labelAnnotations;
+    for (var i = 0; i <= 2; i++) {
+      if (labels[i] && labels[i].description) {
```

Comment:
Use something like this?
`topLabels = labels.map(x => x.description).slice(0,3);`

Reply:
Done.